tests/DCPS/XTypes/XTypes.h

```diff
@@ -28,5 +28,36 @@ enum AdditionalFieldValue {
 const std::string STRING_26 = "abcdefghijklmnopqrstuvwxyz";
 const std::string STRING_20 = "abcdefghijklmnopqrst";
 
+template<typename T>
+bool get_topic(T ts, const DomainParticipant_var dp, const std::string& topic_name,
+  Topic_var& topic, const std::string& registered_type_name)
+{
+  if (ts->register_type(dp, registered_type_name.empty() ? "" : registered_type_name.c_str()) != DDS::RETCODE_OK) {
+    ACE_ERROR((LM_ERROR, "ERROR: register_type failed\n"));
+    return false;
+  }
+  CORBA::String_var type_name = (registered_type_name.empty() ? ts->get_type_name() : registered_type_name.c_str());
```

Comment:
This is not addressing the leak found by valgrind/asan.

When using the `?:` operator the compiler needs to come up with a common type between the two expressions, and it's not going to do what we want it to.  

The usual rule for using `_var` for memory management applies: the result of an allocating call (`get_type_name()` in this case) should be directly put in a `_var`.